enigma/src/main/java/org/quiltmc/enigma/api/stats/StatsGenerator.java

```diff
@@ -291,10 +307,12 @@ public StatsResult getStats(ClassEntry entry) {
 		return this.result.getStats().get(entry);
 	}
 
-	private void update(StatType type, Map<StatType, Integer> mappable, Map<StatType, Map<String, Integer>> unmapped, Entry<?> entry) {
+	private void update(StatType type, Map<StatType, Integer> mappable, Map<StatType, Map<String, Integer>> unmapped, Entry<?> entry, GenerationParameters parameters) {
 		if (this.project.isRenamable(entry)) {
-			if (this.project.isObfuscated(entry) && !this.project.isSynthetic(entry)
-					|| this.fallbackNameProposerIdCache.contains(this.project.getRemapper().getMapping(entry).sourcePluginId())) { // fallback proposed mappings don't count
+			boolean obf = this.project.isObfuscated(entry);
+
+			if ((obf && (parameters.includeSynthetic() || !this.project.isSynthetic(entry)))
```

Comment:
calls to update are always preceded by a synthetic check (except for classes), so this shouldn't be necessary